### PR TITLE
kucoin - networks unification

### PR DESCRIPTION
**Pull request by ttodua**

### CODE REVIEW

**Comment by ttodua** [comment]
ready

File: ts/src/kucoin.ts [hunk 1]
```diff
@@ -847,32 +1151,27 @@ export default class kucoin extends Exchange {
         return result;
     }
 
-    async fetchTransactionFee (code: string, params = {}) {
```
Comment (carlosmiei) [inline ts/src/kucoin.ts:850]: why did u remove the type here?

File: ts/src/kucoin.ts [hunk 2]
```diff
@@ -749,8 +1055,7 @@ export default class kucoin extends Exchange {
          * @method
          * @name kucoin#fetchCurrencies
          * @description fetches all available currencies on an exchange
-         * @see https://docs.kucoin.com/#get-currencies
```
Comment (carlosmiei) [inline ts/src/kucoin.ts:752]: same why removing the endpoint from the jsdoc?

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
p kucoin fetchDepositWithdrawFee "USDT"
Python v3.10.9
CCXT v4.0.9
kucoin.fetchDepositWithdrawFee(USDT)
{'deposit': {'fee': None, 'percentage': None},
 'info': {'availableAmount': '482.6598',
          'chain': 'ERC20',
          'currency': 'USDT',
          'innerWithdrawMinFee': '0',
          'isWithdrawEnabled': True,
          'limitBTCAmount': '0.65278672',
          'limitQuotaCurrencyAmount': '20000.00000000',
          'lockedAmount': '0',
          'precision': 6,
          'quotaCurrency': 'USDT',
          'reason': None,
          'remainAmount': '20000.000000',
          'usedBTCAmount': '0.00000000',
          'usedQuotaCurrencyAmount': '0',
          'withdrawMinFee': '25',
          'withdrawMinSize': '40'},
 'networks': {'ERC20': {'deposit': {'fee': None, 'percentage': None},
                        'withdraw': {'fee': 25.0, 'percentage': False}}},
 'withdraw': {'fee': 25.0, 'percentage': False}}
```

```
p kucoin fetchDepositWithdrawFee "USDT" '{"network":"TRC20"}'
Python v3.10.9
CCXT v4.0.9
kucoin.fetchDepositWithdrawFee(USDT,{'network': 'TRC20'})
{'deposit': {'fee': None, 'percentage': None},
 'info': {'availableAmount': '482.6598',
          'chain': 'TRC20',
          'currency': 'USDT',
          'innerWithdrawMinFee': '0',
          'isWithdrawEnabled': True,
          'limitBTCAmount': '0.65278672',
          'limitQuotaCurrencyAmount': '20000.00000000',
          'lockedAmount': '0',
          'precision': 6,
          'quotaCurrency': 'USDT',
          'reason': None,
          'remainAmount': '20000.000000',
          'usedBTCAmount': '0.00000000',
          'usedQuotaCurrencyAmount': '0',
          'withdrawMinFee': '1',
          'withdrawMinSize': '3'},
 'networks': {'TRC20': {'deposit': {'fee': None, 'percentage': None},
                        'withdraw': {'fee': 1.0, 'percentage': False}}},
 'withdraw': {'fee': 1.0, 'percentage': False}}
```

```
 p kucoin fetchDepositAddress USDT '{"network":"TRC20"}'      
Python v3.10.9
CCXT v4.0.9
kucoin.fetchDepositAddress(USDT,{'network': 'TRC20'})
{'address': 'TMXXX1',
 'currency': 'USDT',
 'info': {'address': 'TMXXX1',
          'chain': 'TRC20',
          'memo': ''},
 'network': 'TRC20',
 'tag': None}
```

```
 p kucoin fetchDepositAddressesByNetwork "USDT"                                       
Python v3.10.9
CCXT v4.0.9
kucoin.fetchDepositAddressesByNetwork(USDT)
{'ERC20': {'address': '0xXXXc',
           'currency': 'USDT',
           'info': {'address': '0xXXXc',
                    'chain': 'ERC20',
                    'contractAddress': '0xdac173d831ec7',
                    'memo': ''},
           'network': 'ERC20',
           'tag': None},
 'SOL': {'address': '6mXXXXY8FDJU',
         'currency': 'USDT',
         'info': {'address': '6mXXXXY8FDJU',
                  'chain': 'SOL',
                  'contractAddress': 'XXXX',
                  'memo': ''},
         'network': 'SOL',
         'tag': None},
 'TRC20': {'address': 'TMXXX1',
           'currency': 'USDT',
           'info': {'address': 'TMXXX1',
                    'chain': 'TRC20',
                    'contractAddress': 'TR7NHqjeKQxGTCi8q8ZY4pL8otSzgjLj6t',
                    'memo': ''},
           'network': 'TRC20',
           'tag': None}}
```

```
p kucoin fetchDeposits USDT None 10
Python v3.10.9
CCXT v4.0.9
kucoin.fetchDeposits(USDT,None,10)
[{'address': 'TMXXX1',
  'addressFrom': None,
  'addressTo': 'TMXXX1',
  'amount': 2500.0,
  'comment': 'Deposit',
  'currency': 'USDT',
  'datetime': '2023-05-03T14:29:31.000Z',
  'fee': {'cost': 0.0, 'currency': 'USDT', 'rate': 0.0},
  'id': None,
  'info': {'address': 'TMXXX1',
           'amount': '2500.00000000',
           'arrears': False,
           'chain': 'trx',
           'createdAt': 1683124171000,
           'currency': 'USDT',
           'fee': '0.00000000',
           'isInner': False,
           'memo': '',
           'remark': 'Deposit',
           'status': 'SUCCESS',
           'updatedAt': 1683124175000,
           'walletTxId': '46f706da619c'},
  'network': 'TRC20',
  'status': 'ok',
  'tag': None,
  'tagFrom': None,
  'tagTo': None,
  'timestamp': 1683124171000,
  'txid': '46f734c490f08b4ae',
  'type': 'deposit',
  'updated': 1683124175000},
```

**Review by carlosmiei — APPROVED** [review]

File: ts/src/kucoin.ts (hunk 1 above)
Comment (ttodua) [reply ts/src/kucoin.ts]: @carlosmiei  sorry, I was away and returned atm,  couldnt reply earlier. so the type removal from method signatures and some other incompleteness were due to the reason that this PR was quite old and I've just migrated them into new kucoin.ts with quick copy-pasting and missed those details sorry :( 
would pay more attention to such cases next times

**Comment by ttodua** [comment]
may I ask about this change - https://github.com/ccxt/ccxt/commit/d375e81bab6bb393452d08e8c7d4eca425a2a4d3#diff-691fa95fe602b9abc71cdc711533d0984458eaea142774a17bf722f20b5980ecR1172 
why lowercasing is neeed? the CodeToId is meant to exactly convert to the needed exchange value. did you remind which cases required to be lowercase?

**Pull request merged**